wiki/start/a-simple-game.md

```diff
@@ -5,605 +5,1035 @@ redirect_from:
+        bucketSprite.translateX(-speed * delta); // move the bucket left
+    }
+}
+```
 
-      ... more to come here ...
-   }
+Run the game again to make sure you can move the bucket left and right with the keyboard.
+
+### Mouse and Touch Controls
+Mouse and Touch controls are related. To react to the user clicking or tapping the screen, call the following method:
+
+```java
+private void input() {
+    float speed = 4f;
+    float delta = Gdx.graphics.getDeltaTime();
+
+    if (Gdx.input.isKeyPressed(Input.Keys.RIGHT)) {
+        bucketSprite.translateX(speed * delta);
+    } else if (Gdx.input.isKeyPressed(Input.Keys.LEFT)) {
+        bucketSprite.translateX(-speed * delta);
+    }
+
+    if (Gdx.input.isTouched()) { // If the user has clicked or tapped the screen
+        // todo:React to the player touching the screen
+    }
+}
 ```
 
-The arguments for `ScreenUtils.clear(r, g, b, a)` are the red, green, blue and alpha component of that color, each within the range [0, 1].
+Now the player has clicked the screen, but where did they click? We can use the methods Gdx.input.getX() and Gdx.input.getY() for this. Unfortunately, these values are in window coordinates which don't correlate to our selected pixels per meter. The coordinates are also upside down because Window coordinates start from the top left. We need to declare a Vector2 object to do some math.
 
-Next we need to tell our camera to make sure it is updated. Cameras use a mathematical entity called a matrix that is responsible for setting up the coordinate system for rendering. These matrices need to be recomputed every time we change a property of the camera, like its position. We don't do this in our simple example, but it is generally a good practice to update the camera once per frame:
+```java
+public class Main implements ApplicationListener {
+    ...
+    Vector2 touchPos;
+```
+
+Initialize the Vector2:
 
 ```java
-   camera.update();
+@Override
+public void create() {
+    ...
+
+    touchPos = new Vector2();
+}
 ```
 
-Now we can render our bucket:
+Notice that we have created a single instance variable for the Vector2 instead of creating it locally. By reusing this Vector2, we prevent the game from triggering the garbage collector frequently which causes lag spikes in the game. This is how we use the Vector2 to move the bucket:
 
 ```java
-   batch.setProjectionMatrix(camera.combined);
-   batch.begin();
-   batch.draw(bucketImage, bucket.x, bucket.y);
-   batch.end();
+private void input() {
+    float speed = 4f;
+    float delta = Gdx.graphics.getDeltaTime();
+
+    if (Gdx.input.isKeyPressed(Input.Keys.RIGHT)) {
+        bucketSprite.translateX(speed * delta);
+    } else if (Gdx.input.isKeyPressed(Input.Keys.LEFT)) {
+        bucketSprite.translateX(-speed * delta);
+    }
+
+    if (Gdx.input.isTouched()) {
+        touchPos.set(Gdx.input.getX(), Gdx.input.getY(), 1); // Get where the touch happened on screen
```

Comment:
touchPos is Vector2, so you no longer need Z coordinate (and it won't compile).
Sorry, i didn't deeply checked the whole nice work you did, but it poped to my eyes.
Cheers.

Reply:
That is a really good catch. I missed that one. Thanks!